internal/pkg/template/workload.go

```diff
@@ -456,6 +491,17 @@ type NetworkLoadBalancer struct {
 	CertificateRequired bool
 }
 
+// ApplicationLoadBalancer holds configuration that's needed for an Application Load Balancer.
+type ApplicationLoadBalancer struct {
+	PublicSubnetCIDRs []string
+	Listener          []ApplicationLoadBalancerRoutineRule
+	HostedZoneAliases AliasesForHostedZone
+	HTTPRedirect      bool
+	HTTPSListener     bool
+	ALBEnabled        bool
```

Comment:
Why do we need this boolean? would the client set `ALB` to `nil` if it's not enabled?

Reply:
I think I was trying to make this parameter as a part of `ALB` but later realized it is used on much higher context such as in `envControllerParameters` and so on. So that is why decided to keep it high level param inside `template.WorkloadOpts` and forgot to remove from this place. I will remove it.